### PR TITLE
chore(windows): move from Makefile to build.sh

### LINKED ISSUE #11316: chore(windows): Consolidate build scripts on windows
We should have a set of common parameters across all build.sh scripts, and consistent ways of discovering platform-specific parameters. See parent #6268  for the attached design document.

### PR DESCRIPTION
**Pull request by mcdurdin**
Fixes #11316.

Now builds from a clean repo:

```
windows/src/build.sh configure build test publish
```

* Cleanup of various build scripts and dependencies.

TODO:
- [x] Move Developer-related build script changes to #11421
- [x] Re-review all build scripts for consistency
- [x] CI linkage - release build
- [x] CI linkage - test build
- [x] user test spec
- [ ] CI linkage - test osk render build --> this will come in a follow-up PR (lower priority)
- [ ] convert remaining makefiles for manual tests and support projects --> this will come in a follow-up PR (lower priority)

# User Testing

* **TEST_WINDOWS:** Please perform a basic regression test of Keyman for Windows to ensure that all functions continue to operate as expected.

### CODE REVIEW

**Comment by mcdurdin** [comment]
Note: split out a number of common changes into separate PRs #11477, #11478, and moved Developer-related fixes into #11421.

**Comment by dinakaranr** [comment]
# Test Results

* **TEST_WINDOWS (PASSED):**  I tested this issue with the attached "Keymandeveloper-18.0.39-alpha-test-11461" build on the Windows 11 OS environment: Here is my observation.
I have followed the acceptance scenario from the help file. I executed the below scenarios

1. Installed the "keyman-18.0.39.exe" file.
2. Open the configuration dialog by clicking the keyman icon from the toolbox tray.
3. Installed keyboards that are required for testing.

SUITE_BASELINE: Keyman for Windows Base Line Tests: Passed
SUITE_CAPSLOCK: Passed
CapsAlwaysOff: Passed
SHIFT: CapsOnOnly/ShiftFreesCaps: Passed
SUITE_TSF_APP: Passed
SUITE_IMX_KEYBOARDS: Passed
SUITE_STORE_AND_CONTEXT: Passed

I followed the test scenario described above test suite. I executed as per the test case instructions and outputs were correctly displayed in "Text Editor". It works well. Thank you.

File: windows/src/engine/inst/build.sh [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+source "$KEYMAN_ROOT/resources/shellHelperFunctions.sh"
+
+builder_describe "Installation files for Keyman Engine for Windows" \
+  clean configure build test publish
+
+# NOTE: not using deps here because we will only do this in the 'publish' phase
+# after all other builds complete
+
+builder_describe_outputs \
+  publish       /windows/src/desktop/inst/keymanengine.msm
```
Comment (rc-swag) [inline windows/src/engine/inst/build.sh:17]: This should either be 
`$KEYMAN_ROOT/windows/release/${VERSION}/keymanengine.msm`
of the local version before copy_installer
`keymanengine.msm`
Either case nothing is copying it to where this line suggests.
Also should see comment below

File: windows/src/engine/inst/build.sh [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+  #
+  wrap-signcode //d "Keyman Engine for Windows" keymanengine.msm
+
+  copy-installer
+}
+
+function copy-installer() {
+  builder_heading copy-installer
+  mkdir -p "$KEYMAN_ROOT/windows/release/${VERSION}"
+  cp keymanengine.msm "$KEYMAN_ROOT/windows/release/${VERSION}/keymanengine.msm"
```
Comment (rc-swag) [inline windows/src/engine/inst/build.sh:54]: Do we no longer include version in the name `keymanengine-${VERSION}.msm

File: windows/src/engine/keyman32/build.sh [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+  create-windows-output-folders
+  build_version.res
+  vs_msbuild keyman32.vcxproj //t:Build "//p:Platform=Win32"
+  cp "$WIN32_TARGET" "$WINDOWS_PROGRAM_ENGINE"
+  cp "$WIN32_TARGET_PATH/keyman32.pdb" "$WINDOWS_DEBUGPATH_ENGINE"
+}
+
+function do_publish() {
+  wrap-signcode //d "Keyman Engine for Windows" "$WINDOWS_PROGRAM_ENGINE/keyman32.dll"
+  wrap-symstore "$WINDOWS_PROGRAM_ENGINE/keyman32.dll" //t keyman-engine-windows
```
Comment (rc-swag) [inline windows/src/engine/keyman32/build.sh:41]: Just querying this change in behaviour 
for my local publish this gets skipped as $KEYMAN_SYMSTOREPATH is not is set an so wrap-symstore exits early. Previously the Defines.mak file had this line.
# KEYMAN_SYMSTOREPATH defaults to sibling folder "symbols". If it is not present,
# then we won't attempt to write symbols to the store.
```
!IFNDEF KEYMAN_SYMSTOREPATH
KEYMAN_SYMSTOREPATH=$(KEYMAN_ROOT)\..\symbols
!ENDIF
```

File: windows/src/engine/inst/build.sh (hunk 1 above)
Comment (mcdurdin) [reply windows/src/engine/inst/build.sh:17]: Good catch. Fixed.

File: windows/src/engine/inst/build.sh (hunk 2 above)
Comment (mcdurdin) [reply windows/src/engine/inst/build.sh:54]: Yeah, another good catch. The Keyman Desktop .wxs has:

https://github.com/keymanapp/keyman/blob/53b583e009f37803f84d3aace38cd9813dafaed5/windows/src/desktop/inst/keymandesktop.wxs#L61

Which is why the build worked. Fixed this one too.

File: windows/src/engine/keyman32/build.sh (hunk 3 above)
Comment (mcdurdin) [reply windows/src/engine/keyman32/build.sh:41]: I will add a check to wrap-symstore.

Comment (mcdurdin) [reply windows/src/engine/keyman32/build.sh:41]: It's already there, see:

https://github.com/keymanapp/keyman/blob/8b2070c339a0ec424ae5d7a9fcbb23e97d8afc6a/resources/build/win/environment.inc.sh#L117-L121

**Review by rc-swag — APPROVED** [review]
lgtm

**Comment by keyman-server** [comment]
Changes in this pull request will be available for download in [Keyman version 18.0.42-alpha](https://keyman.com/downloads/releases/alpha/18.0.42)

**Pull request merged**